linkerd/app/outbound/src/http/concrete.rs

```diff
@@ -69,24 +52,17 @@ impl<N> Outbound<N> {
                         .layer(stack_labels("http", "endpoint")),
                 )
                 .instrument(|e: &Endpoint| info_span!("endpoint", addr = %e.addr))
-                // Resolve the service to its endpoints and balance requests over them.
-                //
-                // We *don't* ensure that the endpoint is driven to readiness here, because this
-                // might cause us to continually attempt to reestablish connections without
-                // consulting discovery to see whether the endpoint has been removed. Instead, the
-                // endpoint stack spawns each _connection_ attempt on a background task, but the
-                // decision to attempt the connection must be driven by the balancer.
-                //
-                // TODO(ver) remove the watchdog timeout.
-                .push(resolve::layer(resolve, config.discovery_idle_timeout * 2))
-                .push_on_service(http::balance::layer(
-                    crate::EWMA_DEFAULT_RTT,
-                    crate::EWMA_DECAY,
-                ))
-                .check_make_service::<Concrete, http::Request<http::BoxBody>>()
-                .push(svc::MapErr::layer(Into::into))
+                .check_new_service::<Endpoint, http::Request<_>>()
+                .push_new_clone()
+                .check_new_new::<Concrete, Endpoint>()
+                .check_new_new_service::<Concrete, Endpoint, http::Request<_>>()
+                .push(endpoint::NewFromMetadata::layer(config.inbound_ips.clone()))
+                .check_new_new_service::<Concrete, (_, _), http::Request<_>>()
+                .push(http::NewBalance::layer(resolve))
+                .check_clone()
+                .check_new::<Concrete>()
+                .check_new_service::<Concrete, http::Request<_>>()
```

Comment:
similarly, isn't this equivalent to just
```suggestion
                .check_new_service::<Concrete, http::Request<_>>()
```

Reply:
i'm going to clean this up before review (see TODOs in description), but this differentiates whether a compile error is on the new or the service.